From 7fe1f97a128be5f48dcd1232250e96bd581a35c1 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Tue, 1 May 2012 18:42:12 +0200
Subject: [PATCH] configure: Reset LIBS after ELF check

If action-if-found is not specified, AC_CHECK_LIB will append the library
to LIBS. As we don't want to link everything against libelf, reset LIBS
after doing the checks.
---
 configure.ac |    2 ++
 1 file changed, 2 insertions(+)

Index: glib-2.32.2/configure.ac
===================================================================
--- glib-2.32.2.orig/configure.ac	2012-05-01 20:00:42.839506415 +0200
+++ glib-2.32.2/configure.ac	2012-05-01 20:00:42.867506416 +0200
@@ -1821,10 +1821,12 @@
 PKG_CHECK_MODULES([LIBELF], [libelf], [have_libelf=yes], [have_libelf=maybe])
 
 if test $have_libelf = maybe; then
+  glib_save_LIBS=$LIBS
   AC_CHECK_LIB([elf], [elf_begin], , [have_libelf=no])
   AC_CHECK_LIB([elf], [elf_getshdrstrndx], , [have_libelf=no])
   AC_CHECK_LIB([elf], [elf_getshdrnum], , [have_libelf=no])
   AC_CHECK_HEADER([libelf.h], , [have_libelf=no])
+  LIBS=$glib_save_LIBS
 
   if test $have_libelf != no; then
     LIBELF_LIBS=-lelf
